mopidy/core/library.py

```diff
@@ -152,12 +168,18 @@ def get_images(self, uris):
 
         results = {uri: tuple() for uri in uris}
         for backend, future in futures.items():
-            try:
+            with _backend_error_handling(backend):
+                if future.get() is None:
+                    continue
+                validation.check_instance(future.get(), collections.Mapping)
                 for uri, images in future.get().items():
+                    if uri not in uris:
+                        name = backend.actor_ref.actor_class.__name__
+                        logger.warning(
+                            '%s backend returned image for URI we did not '
+                            'ask for: %s', name, uri)
```

Comment:
Can you just raise a ValdiationError here and have _backend_error_handling take care of the logging? You'll loose the entire result from that backend, but that's just punishment for giving us bloated results.